Added package events to logistics peripherals

BirbIrl opened the pull request:
Packager:
`package_created` - when the packager makes a package, return the package object
`package_received`- when the packager receieves a package, returns the package object

Repackager:
`package_repackaged` - when the repackager repackages, returns the package object and count (number)
`package_received` - when the repackager receives a package, returns the package object


Forgport:
`package_sent` - when the forgport sends a package on the chain conveyor, returns the package object
`package_recieved` - when the forgport recieves a package from the chain conveyor, returns the package object

Postbox:
`package_sent` - when the forgport sends a package onto a train, returns the package object
`package_recieved` - when the forgport recieves a package from a train, returns the package object

ElementW commented:
LGTM, but there are a handful of lines that are touched yet the code itself is unchanged, maybe you could unstage them before committing? Just in case they end up conflicting with upstream create in the meantime for no good reason

BirbIrl committed:
Added package events to logistics peripherals

events added to the packager, repackager, frogport and postbox

rebased and cleaned

BirbIrl commented:
> LGTM, but there are a handful of lines that are touched yet the code itself is unchanged, maybe you could unstage them before committing? Just in case they end up conflicting with upstream create in the meantime for no good reason

aye my bad lol. I was hoping --ignore-whitespace would be enough and forot to re-check if it's fine since i've made the pr whoopsies

ElementW commented:
Honestly I was expecting the integration to require more code but then again I haven't actually used Create since the logistics update came out so :woman_shrugging: 
Merging

BirbIrl commented:
> Honestly I was expecting the integration to require more code but then again I haven't actually used Create since the logistics update came out so 🤷‍♀️ Merging

the original logistics integration was like 1.5k lines of code https://github.com/Creators-of-Create/Create/pull/7883

Thanks to your library it was as simple as slapping on `thingDoer.DoThing()`  whenever a package was made/received/sent by any of the logistics blocks to add the event system :P
